zerver/lib/push_notifications.py

```diff
@@ -1057,6 +1063,10 @@ def get_apns_alert_subtitle(
         return _("{full_name} mentioned everyone:").format(full_name=sender_name)
     elif message.recipient.type == Recipient.PERSONAL:
         return ""
+    elif message.recipient.type == Recipient.DIRECT_MESSAGE_GROUP:
+        recipients = get_display_recipient(message.recipient)
+        if len(recipients) <= 2:
+            return ""
```

Comment:
since we're not actually using the `recipients` besides checking length, I wonder if it wouldn't be fast to fetch the `DirectMessageGroup` for the `message.recipient` and just look at `group_size`? on the other hand `get_display_recipient` is cached and already called in other places in this codepath, so maybe it's good enough or actually better.

So all in all, I think this can stay the way it is, but just noting in case someone has other thoughts on this.

Reply:
Yeah I had the same thought initially -- but because `get_display_recipient` uses memcached, it's actually faster this way.

Reply:
Discussed here: https://github.com/zulip/zulip/pull/34864#discussion_r2141473531

Thanks for flagging though!